app/messaging/wait-for-idle-subscription.js

```diff
@@ -0,0 +1,28 @@
+const { MessageReceiver } = require('ffc-messaging')
+const Long = require('long')
+const config = require('../config/message')
+const sleep = require('./sleep')
+
+const waitForIdleSubscription = async (subscription) => {
+  let receiver
+  try {
+    receiver = new MessageReceiver(subscription)
+    let idle = false
+    do {
+      const messages = await receiver.peekMessages(config.idleCheckBatchSize, { fromSequenceNumber: Long.fromInt(0) })
```

Comment:
The intention here is to make sure that the message subscription in question is empty and if not wait for it to be empty.

`fromSequenceNumber` is needed because otherwise each `peekMessages` call would retrieve progressive batches of 250.

eg. first iteration of loop calls `0-249`, second iteration calls `250-499` etc.

So for the call after the last set of messages it would report `0` messages in the queue causing this loop to exit prematurely.

So starting from 0 means it will always return the first 250 messages.

The rationale for allowing messages that have already been delivered at least once is to prevent "poison" messages that have made it onto our queues that are stuck in the `2000` redelivery attempt loop from blocking statement generation for five days.

See below for why we're doing this in the first place.